### PR TITLE
Rnorm

### PR DESCRIPTION
**Pull request by nataliecrowe060**
My Saturday work- both some tests in another file and the main markdown file with an added histogram/rnorm run and a change to the sensitivity analysis.

### CODE REVIEW

File: Methane_MarkDown.Rmd [hunk 1]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+x<- 341
+firstValue <- x * (1 - 0.2)
+lastValue <- x * (1 + 0.2)
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), seq(firstValue, lastValue, 5))
+```
+
+Once we have our differing parameter values we can create a graph. Using the graph function `facet_wrap`to separate the results based on variable.
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:150]: "Once we have our model outputs we can..."

File: Methane_MarkDown.Rmd [hunk 2]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+
+Then we tell hector to run a range of values. In this case we ask for 20% above and bellow the set value of `NATURAL_CH4`(341Tg) at intervals of 5. 
+
+```{r sen-param}
+run_with_param_range <- function(core, parameter, values) {
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+x<- 341
+firstValue <- x * (1 - 0.2)
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:145]: `ch4_low` and `ch4_high` would be more descriptive names.

File: Methane_MarkDown.Rmd [hunk 3]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+```{r}
+x <- 341*0.2
+data<- data.frame(value= rnorm(1000, 341, x))
+```
+
+Then we can use `ggplot2` to create a histogram with the randomized values.
+```{r}
+ggplot(data, aes(x=value))+
+  geom_histogram(binwidth = 1)+
+  labs( x= "Natural CH4 Level", y= "count", main= "Natural CH4 Histogram")
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:174]: What are the units of the x axis?

File: Methane_MarkDown.Rmd [hunk 4]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
          color = "Units") + facet_wrap(~variable)
 ```
 ... to show our data side-by-side.
+
+## Sensitivity Analysis
+
+To examine SSP245's Natural methane  sensitivity we first ask hector to fetch the data. We then create a function to run our parameter set to a particular value. In this case we set our value at `358.05`- twenty percent higher than the original value.
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:115]: >we first ask hector to fetch the data

What data? 😕

File: Methane_MarkDown.Rmd [hunk 5]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+
+```{r call-for-param}
+core <- newcore(ini_file)
+run(core)
+paramr<- fetchvars(core, NA, NATURAL_CH4())
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:120]: Maybe use a more descriptive name, e.g. `default`?

File: Methane_MarkDown.Rmd [hunk 6]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+
+```{r sen-param}
+run_with_param_range <- function(core, parameter, values) {
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+x<- 341
+firstValue <- x * (1 - 0.2)
+lastValue <- x * (1 + 0.2)
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), seq(firstValue, lastValue, 5))
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:147]: You may want to do
```r
ch4_seq <- seq(firstValue, lastValue, 5)
sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), ch4_seq)
```
...and then use `ch4_seq` for your graph below.

File: Methane_MarkDown.Rmd [hunk 7]
```diff
@@ -111,3 +109,68 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+  aes(x = year, y = value, color = parameter_value, group = parameter_value) +
+  geom_line() +
+  facet_wrap(~variable, scales = "free_y") +
+  labs(color = "NATURAL CH4") +
+  scale_color_viridis_c() 
+```
+
+##Adding a Histogram 
+
+Using the same `Natural_CH4` data, we can also make a histogram plot. First we use `rnorm` to set a `mean=` to the constant( 341Tg ) and a `sd` of the constant multiplied by `0.2`.
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:163]: Since you say "using the same data", I would use `ch4_seq` (see above) for your graph.

**Review by bpbond — CHANGES_REQUESTED** [review]
Nice work @nataliecrowe060

**Review by leeyap — APPROVED** [review]
Great job!

**Pull request merged**